aiidalab/app.py

```diff
@@ -320,9 +316,10 @@ def current_revision(self):
 
             return None  # current revision not on the release line
 
-    def __init__(self, name, app_data, aiidalab_apps_path, watch=True):
+    def __init__(self, name, watch=True):
         super().__init__()
 
+        app_data = load_app_registry()['apps'].get(name)
```

Comment:
Just to mention. Initially the idea was to load `app_data` only once to reduce the time needed. But, with caching enabled, everything seems to be fast here.

Reply:
Got it. I think the lesson here is to ensure to optimize at the right spot and also not prematurely. Code logic and clarity usually trumps micro optimizations.